install_config/persistent_storage/persistent_storage_glusterfs.adoc

```diff
@@ -13,567 +14,208 @@ toc::[]
+with {product-title}. {product-title} comes with a built-in GlusterFS volume
+driver, allowing it to make use of existing volumes on existing clusters. See
+xref:provisioning[provisioning] for more on how to make use of existing volumes.
+
+For {gluster-native} and {gluster-external}, it is recommended to use the
+xref:../../install/advanced_install.adoc[Advanced Installer] to install the
+required components.
+
+[[install-gluster-nodes]]
+=== {gluster-external}: Installing {gluster} Nodes
+
+include::install_config/persistent_storage/topics/glusterfs_install_nodes.adoc[]
+
+[[install-advanced-installer]]
+=== Using the Advanced Installer
+
+The xref:../../install/advanced_install.adoc[Advanced Installer] can be used to
+install one or both of two GlusterFS clusters:
+
+- `glusterfs`: A general storage cluster for use by user applications.
+- `glusterfs-registry`: A dedicated storage cluster for use by infrastructure
+applications such as an integrated OpenShift Container Registry.
+
+Both of these are defined in the inventory hosts file.
+
+The definition of the clusters is done by including the relevant names in the
+`[OSEv3:children]` group, creating similarly named groups, and then populating
+the groups with the node information. The clusters can then be configured
+through a variety of variables in the `[OSEv3:vars]` group. `glusterfs`
+variables begin with `openshift_storage_glusterfs_` and `glusterfs-registry`
+variables begin with `openshift_storage_glusterfs_registry_`. A few other
+variables, such as `openshift_hosted_registry_storage_kind`, interact with the
+GlusterFS clusters.
+
+It is recommended to specify version tags for all containerized components.
+This is primarily to prevent components, particularly the {gluster} pods, from
+upgrading after an outage which may lead to a cluster of widely disparate
+software versions. The relevant variables are:
+
+- `openshift_storage_glusterfs_version`
+- `openshift_storage_glusterfs_block_version`
+- `openshift_storage_glusterfs_s3_version`
+- `openshift_storage_glusterfs_heketi_version`
+- `openshift_storage_glusterfs_registry_version`
+- `openshift_storage_glusterfs_registry_block_version`
+- `openshift_storage_glusterfs_registry_s3_version`
+- `openshift_storage_glusterfs_registry_heketi_version`
+
+For a complete list of variables, see the
+link:{gluster-role-link}[GlusterFS role README] on GitHub.
+
+Once the variables are configured, there are several playbooks available
+depending on the circumstances of the installation:
+
+- The main playbook of the Advanced Installer can be used to deploy the
+GlusterFS clusters in tandem with an initial installation of {product-title}.
+  ** This includes deploying an integrated OpenShift Container Registry that
+  uses GlusterFS storage.
+  ** This does not include OpenShift Logging or OpenShift Metrics, as that is
+  currently still a separate step. See link:install-example-infra[{gluster-native} for OpenShift Logging and Metrics]
```

Comment:
Why does this have to be a separate step? If gluster is deployed, and is provisioned as the default storage class, and logging/metrics are configured to use dynamic provisioning - using the default storage class (or a defined storage class) then it should be possible to connect to the `glusterfs` the `glusterfs-registry` depending on which storage class is used in the configuration.

Reply:
When I last tried to do it all in one step there were some issues I encountered that could not be fixed in time for OCP 3.9. Indeed, I am hoping to get it all as a single step for OCP 3.10.

Reply:
We do this for internal labs(quicklab) so I know it's possible. If it breaks with 3.9 then it should be document as a 3.9 known issue, and we should be working to get out a z stream fix.

Reply:
I have not heard of anyone successfully deploying Logging and Metrics during initial installation with the advanced installer onto a gluster-block StorageClass that was also created during that same initial installation. This was broken back when I tried it in 3.7 and again when I tried it in 3.9. If it turns out I did something wrong then that's great news, but unfortunately that's not what we have had CNS and OCP QE verify, and it is not what we have been telling others to do. I would leave it as-is for now and then if we find it does work we'll get it fully tested and THEN update the documentation.